fighters/robot/src/status/special_hi.rs

```diff
@@ -173,12 +229,19 @@ unsafe extern "C" fn special_hi_main_loop(fighter: &mut L2CFighterCommon) -> L2C
     KineticModule::enable_energy(fighter.module_accessor, *FIGHTER_KINETIC_ENERGY_ID_MOTION);
 
     let sfx =
-        if charge_frame >= 20.0 { "se_common_bomb_l" }
+        if charge_frame >= 40.0 { "se_common_bomb_ll" }
+        else if charge_frame >= 25.0 { "se_common_bomb_l" }
         else if charge_frame >= 10.0 { "se_common_bomb_m" }
         else { "se_common_bomb_s" };
 
     if charge_frame >= 10.0 {
-        launch_speed.y = ((1.65 + (0.05 * charge_frame)) - (0.025 * rot_x.abs())).min(3.75);
+        if fighter.is_situation(*SITUATION_KIND_GROUND) {
+            launch_speed.x = 0.30 * rot_x.abs() * (((charge_frame * 2.0) - 18.0).clamp(0.0, 32.0) / 32.0);
+            launch_speed.y = ((1.65 + (0.10 * charge_frame)) - (0.035 * rot_x.abs())).min(3.75);
+        } else {
+            launch_speed.x = 0.15 * rot_x.abs() * (((charge_frame) - 18.0).clamp(0.0, 32.0) / 32.0);
+            launch_speed.y = ((1.65 + (0.05 * charge_frame)) - (0.025 * rot_x.abs())).min(3.75);
+        }
```

Comment:
it's genuinely hard to follow what the launch speeds are supposed to be for different scenarios in this larger function. You're doing a lot of math in-line even though it looks like only a few of these numbers actually change in the different if blocks. You should try to extract common patterns out to earlier in the function so they can be reused without bloating the function.